### DISCUSSION TITLE
How to use LpExtractor

### DISCUSSION THREAD

**Discussion by sqPoseidon** [egraphs-good/egg#214] [Q&A]
Hi, I'm new to Rust and Egg. I got one problem when using LpExtractor following: https://github.com/egraphs-good/egg/blob/4cfd22b7fc0218627d62aec448ed7000580546c9/tests/math.rs#L351
I have installed the coin-cbc via `$ conda install coin-or-cbc`. And in the rust file, `use egg::*` is added. Then I got the error message: 
```
349 |     let lp_best = LpExtractor::new(&runner.egraph, AstSize).solve(root);
    |                   ^^^^^^^^^^^ use of undeclared type `LpExtractor`
```

In the document, I noticed one announcement `Available on crate feature lp only`. Do I need to add something to the `Cargo.toml`? Or do I need to add something else in the `.rs` file?

**Comment by mwillsey** [comment]
Yes, you need to use [features](https://doc.rust-lang.org/cargo/reference/features.html#dependency-features). So something like
```toml
egg = { version = "whatever", features = ["lp"] }
```

**Reply by sqPoseidon** [reply]
Thank you @mwillsey.
